Sources/swift-build/main.swift

```diff
@@ -35,36 +31,55 @@ do {
         try chdir(dir)
     }
     
+    func parseManifest(path path: String, baseURL: String) throws -> Manifest {
+        let bindir = try! Process.arguments.first!.parentDirectory.abspath()
+        let libdir: String
+        if Path.join(bindir, "PackageDescription.swiftmodule").isDirectory { //FIXME HACK
+            libdir = bindir
+        } else {
+            libdir = try! Path.join(bindir, "../lib/swift/pm").abspath()
+        }
+        return try Manifest(path: path, baseURL: baseURL, swiftc: Toolchain.swiftc, libdir: libdir)
+    }
+    
     func fetch(root: String) throws -> [Package] {
-        let manifest = try Manifest(path: root, Manifest.filename, baseURL: root)
-        return try get(manifest)
+        let manifest = try parseManifest(path: root, baseURL: root)
+        return try get(manifest, manifestParser: parseManifest)
     }
 
     switch mode {
-        case .Build(let conf):
-            let dirs = try directories()
-            let packages = try fetch(dirs.root)
-            let (modules, products) = try transmute(packages)
-            let yaml = try describe(dirs.build, conf, modules, products, Xcc: opts.Xcc, Xld: opts.Xld, Xswiftc: opts.Xswiftc)
-            try build(YAMLPath: yaml, target: "default")
+    case .Build(let conf):
+        let dirs = try directories()
+        let packages = try fetch(dirs.root)
+        let (modules, products) = try transmute(packages)
+        let yaml = try describe(dirs.build, conf, modules, products, Xcc: opts.Xcc, Xld: opts.Xld, Xswiftc: opts.Xswiftc)
+        try build(YAMLPath: yaml, target: "default")
+
+    case .Init:
+        try initPackage()
+        
+    case .Fetch:
+        try fetch(try directories().root)
+
+    case .Usage:
+        usage()
 
-        case .Init:
-            try initPackage()
-            
-        case .Fetch:
-            try fetch(try directories().root)
+    case .Clean(.Dist):
+        try rmtree(try directories().root, "Packages")
+        fallthrough
+    case .Clean(.Build):
+        try rmtree(try directories().root, ".build")
 
-        case .Usage:
-            usage()
+    case .Version:
+        Swift.print("Apple Swift Package Manager 0.1")
 
-        case .Clean(.Dist):
-            try rmtree(try directories().root, "Packages")
-            fallthrough
-        case .Clean(.Build):
-            try rmtree(try directories().root, ".build")
+    case .Dump:
+        let dirs = try directories()
+        let packages = try fetch(dirs.root)
+        let (modules, products) = try transmute(packages)
+        let swiftModules = modules.flatMap{ $0 as? SwiftModule }
 
-        case .Version:
-            print("Apple Swift Package Manager 0.1")
+        try Xcodeproj.create(path: try getcwd(), package: packages.last!, modules: swiftModules, products: products)
```

Comment:
Could we please add a way to specify the directory to create the project in? Something like `swift build --xcodeproj --path MyCustomXcodeProjectFolder`. I'd prefer to not pollute the root folder with Xcode projects, especially when many of them will run mostly on Linux servers.